wallet/trezor-common/src/tests/mod.rs

```diff
@@ -0,0 +1,557 @@
+            },
+        ))),
+        8 => chain::TxOutput::IssueNft(
+            primitives::H256(rng.gen()).into(),
+            Box::new(chain::tokens::NftIssuance::V0(
+                chain::tokens::NftIssuanceV0 {
+                    metadata: chain::tokens::Metadata {
+                        creator: if rng.gen::<bool>() {
+                            Some(chain::tokens::TokenCreator {
+                                public_key: crypto::key::PrivateKey::new_from_rng(
+                                    rng,
+                                    KeyKind::Secp256k1Schnorr,
+                                )
+                                .1,
+                            })
+                        } else {
+                            None
+                        },
+                        name: make_random_bytes(rng),
+                        description: make_random_bytes(rng),
+                        ticker: make_random_bytes(rng),
+                        icon_uri: if rng.gen::<bool>() {
+                            Some(make_random_bytes(rng)).into()
+                        } else {
+                            None.into()
+                        },
+                        additional_metadata_uri: if rng.gen::<bool>() {
+                            Some(make_random_bytes(rng)).into()
+                        } else {
+                            None.into()
+                        },
+                        media_uri: if rng.gen::<bool>() {
+                            Some(make_random_bytes(rng)).into()
+                        } else {
+                            None.into()
+                        },
+                        media_hash: make_random_bytes(rng),
+                    },
+                },
+            )),
+            make_random_destination(rng),
+        ),
+        9 => chain::TxOutput::DataDeposit(make_random_bytes(rng)),
+        10 => chain::TxOutput::Htlc(
+            make_random_value(rng),
+            Box::new(chain::htlc::HashedTimelockContract {
+                secret_hash: chain::htlc::HtlcSecretHash(rng.gen()),
+                spend_key: make_random_destination(rng),
+                refund_timelock: make_random_lock(rng),
+                refund_key: make_random_destination(rng),
+            }),
+        ),
+        _ => chain::TxOutput::CreateOrder(Box::new(chain::OrderData::new(
+            make_random_destination(rng),
+            make_random_value(rng),
+            make_random_value(rng),
+        ))),
+    }
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn check_input_encodings(#[case] seed: Seed) {
+    let mut rng = make_seedable_rng(seed);
+
+    let inp = make_random_input(&mut rng);
+
+    let simple_inp: crate::TxInput = inp.clone().into();
+
+    assert_eq!(inp.encode(), simple_inp.encode());
+
+    let decoded_simple_inp =
+        chain::TxInput::decode_all(&mut simple_inp.encode().as_slice()).unwrap();
+    assert_eq!(decoded_simple_inp, inp);
```

Comment:
It's better to perform the checks multiple times (e.g. 100 or 1000) in a loop, to increase the chance of failure if things get broken.
Same for the other test.